instrumentation/powerjob-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/powerjob/v4_0/PowerJobHelper.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.powerjob.v4_0;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import java.util.function.Predicate;
+import tech.powerjob.worker.core.processor.ProcessResult;
+
+public final class PowerJobHelper {
+
+  private final Instrumenter<PowerJobProcessRequest, Void> instrumenter;
+
+  private final Predicate<ProcessResult> failedStatusPredicate;
+
+  private PowerJobHelper(
+      Instrumenter<PowerJobProcessRequest, Void> instrumenter,
+      Predicate<ProcessResult> failedStatusPredicate) {
+    this.instrumenter = instrumenter;
+    this.failedStatusPredicate = failedStatusPredicate;
+  }
+
+  public static PowerJobHelper create(
+      Instrumenter<PowerJobProcessRequest, Void> instrumenter,
+      Predicate<ProcessResult> failedStatusPredicate) {
+    return new PowerJobHelper(instrumenter, failedStatusPredicate);
+  }
+
+  public Context startSpan(Context parentContext, PowerJobProcessRequest request) {
+    if (!instrumenter.shouldStart(parentContext, request)) {
+      return null;
+    }
+    return instrumenter.start(parentContext, request);
+  }
+
+  public void stopSpan(
+      ProcessResult result,
+      PowerJobProcessRequest request,
+      Throwable throwable,
+      Scope scope,
+      Context context) {
+    if (scope == null) {
+      return;
+    }
+    if (failedStatusPredicate.test(result)) {
+      request.setFailed();
+    }
```

Comment:
Usually we'd just pass the result as `response` to the instrumenter then in the span status extractor you could just use `result.isSuccess()` without the need to pass `failedStatusPredicate` around.

Reply:
Done